lib/ddtrace/contrib/action_cable/event.rb

```diff
@@ -60,9 +56,9 @@ def subscription(*args)
           # could leak into the new trace. This "cleans" current context,
           # preventing such a leak.
           def ensure_clean_context!
-            return unless tracer.active_span
+            return unless Datadog::Tracing.active_span
 
-            tracer.provider.context = Context.new
+            Datadog::Tracing.send(:tracer).provider.context = Context.new
           end
```

Comment:
Indeed. It was a hack implemented into integrations to prevent a scenario in which context would never reset and traces would leak. To this end, it was effective, even if terribly ugly.

Since the trace model rework, most of these context reset behaviors are obsolete: when the root span closes, the context auto-clears the active trace, thus preventing the leak. This hack instead becomes a bit heavy handed.

However, there is still a scenario in which it seems to cover us: if the root span is started manually (without a block) but never finished, the condition to auto-finish the trace & reset the context will never be reached. Thus the trace context will think the trace is still active. In a sense it is, or at least it's totally indistinguishable whether it should or shouldn't be.

I'm reticent to embody `reset_context!` into the API first class because it's a heavy handed operation we generally don't want to happen. Given `.send(:tracer)` is already on the path here, I think there's already an expectation that internal access is occurring and thus support is not guaranteed.

I think this is okay for now: I would like to clean this up in 1.x, but I'm a bit afraid if we messed with this now, we could create more problems for the 1.0 release... thus I'd like to defer action on this.

Reply:
I can't force you to do anything, but I'm a bit disappointed that discussing the public api for 1.0 is out of scope in a PR called "implement public tracing API".

I understand not doing it on this PR (it's already quite big), but I do think this is a valuable thing to do for 1.0.0.